src/pymmcore_widgets/control/_stage_explorer/_stage_explorer.py

```diff
@@ -610,6 +687,154 @@ def _t_half_width(self) -> np.ndarray:
         T_center[1, 3] = -self._mmc.getImageHeight() / 2
         return T_center
 
+    # ROIs ------------------------------------------------------------------------
+
+    def _active_roi(self) -> ROIRectangle | None:
+        """Return the next active ROI."""
+        return next((roi for roi in self._rois if roi.selected()), None)
+
+    def _on_mouse_press(self, event: MouseEvent) -> None:
+        """Handle the mouse press event."""
+        canvas_pos = (event.pos[0], event.pos[1])
+
+        picked = None
+        for roi in self._rois:
+            if not picked and (grb := roi.obj_at_pos(event.pos)) is not None:
+                roi.anchor_at(grb, event.pos)
+                roi.set_selected(True)
+                picked = roi
+            else:
+                roi.set_selected(False)
+
+        if self._active_roi() is not None:
+            self._stage_viewer.view.camera.interactive = False
+
+        # (button = 1 is left mouse button)
+        elif self._actions[ROIS].isChecked() and event.button == 1:
+            self._stage_viewer.view.camera.interactive = False
+            # create the ROI rectangle for the first time
+            roi = self._create_roi(canvas_pos)
+            self._undo_stack.push(InsertRoiCommand(self, roi))
+
+    def _create_roi(self, canvas_pos: tuple[float, float]) -> ROIRectangle:
+        """Create a new ROI rectangle and connect its events."""
+        roi = ROIRectangle(self._stage_viewer.view.scene)
+        roi.visible = True
+        roi.set_selected(True)
+        roi.set_anchor(roi._canvas_to_world(canvas_pos))
```

Comment:
i made some additional cleanups on ROI.  Next thing I see that we should do is here: we should pass the canvas position into the creation of the ROIRectangle, so that the very first place that the handles are positioned is correctly in world space.  (i.e. "please create an `ROIRectangle` *at* this position`), to avoid a flash of seeing the handles at the origin, and then quickly moving them when the mouse drags